### PR TITLE
chore(backport): merge release-2.5 to master

### PR DESCRIPTION
**Pull request by bartsmykla**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - no relevant issues
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - it won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - no tests
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - there is no need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label)
  - it is backport

> Changelog: skip

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -239,7 +239,7 @@ jobs:
         description: The CNI networking plugin to use [flannel | calico]
         type: string
         default: flannel
-      deltaKDS:
+      legacyKDS:
```
Comment (slonka) [inline .circleci/config.yml:242]: REMEMBER NOT TO SQUASH :)

**Review by slonka — APPROVED** [review]

**Comment by slonka** [comment]
I think that unit test is failing because of https://github.com/kumahq/kuma/pull/8292#issuecomment-1801887117

**Comment by lahabana** [comment]
Be VERY mindful to not squash merge but to merge with a merge commit (you'll need to reeanble it first).

**Comment by bartsmykla** [comment]
Superseeded by https://github.com/kumahq/kuma/pull/8358

**Pull request closed without merging**